src/main/java/org/springframework/data/solr/core/SolrTemplate.java

```diff
@@ -89,8 +90,9 @@
  * @author Joachim Uhrlass
  * @author Francisco Spaeth
  * @author Shiradwade Sateesh Krishna
+ * @author Venil Noronha
  */
-public class SolrTemplate implements SolrOperations, InitializingBean, ApplicationContextAware {
+public class SolrTemplate implements SolrOperations, MulticoreSolrOperations, InitializingBean, ApplicationContextAware {

```

Comment:
Ok, this seems to be another problem. I think this shall be fixed first. Maybe we can re-think about the current solution. If i ask myself 'why do we need specialized classes for multi-core support?' my answer would be 'to specify the core name'. But is this really requiring all this duplication? Maybe we can use a core name provider which returns the default name in case of single core and the appropriate name in case of multi core? So we would change only the SolrTemplate by the changes in SolrOperations. Maybe we find a way how to provide the name without any change of SolrOperations.